sklearn/neighbors/_classification.py

```diff
@@ -278,6 +299,27 @@ def predict_proba(self, X):
             by lexicographic order.
         """
         if self.weights == "uniform":
+            # TODO: systematize this mapping of metric for
+            # PairwiseDistancesReductions.
+            metric = _adjusted_metric(self.metric, self.p)
+            if (
+                self._fit_method == "brute"
```

Comment:
Here too we need to call `check_is_fitted(self, "_fit_method")` before this condition.

Furthermore, we also need to call the input data validation before calling `ArgKminClassMode.is_usable_for(X, self._fit_X, metric)`. Typically this should amount to something like:

```python
            if self.metric == "precomputed":
                X = _check_precomputed(X)
            else:
                X = self._validate_data(X, accept_sparse="csr", reset=False, order="C")
```

I am not sure if `order="C"` is required or not. If not, let's leave it to avoid a memory copy if the data is provided as F layout.

In `main` this was not needed because input validation is also performed in the `kneighbors` public method.

Reply:
`X` is already validated in `fit` isn't it? It is not obvious to me why we need to include this validation here as well.

Reply:
The `X` in this context is `X_test`.

The `X` you are mentioning is `X_train`, which is stored as `self._fit_X`.

Reply:
🤦 thank you, not sure how I missed it

Reply:
No problem.

Reply:
More problematically, I am not sure how our test suite has missed it. I suppose this is because our common tests only run with the default metric (which is fine) but unfortunately, while the Euclidean case is not implemented we have to branch in predict which delegates the validation to `kneighbors` for Euclidean and currently does not validate for the other metrics (which still needs to be fixed in this PR as far as I can see).